src/classes/QueryPredicate.cls

```diff
@@ -0,0 +1,60 @@
+/*************************************************************************************************
+* This file is part of the Nebula Framework project, released under the MIT License.             *
+* See LICENSE file or go to https://github.com/jongpie/NebulaFramework for full license details. *
+*************************************************************************************************/
+public class QueryPredicate {
```

Comment:
Ok, that makes total sense and seems like a pretty elegant way to handle. But now I'm curious about your implementation - do you still create methods in LeadRepo for your generic/commonly used queries? Or are you just using "get" request (QueryPredicate) for everything?